test/e2e/checkinterval_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultHealthCheckInterval = 5 * time.Second
+	deploymentTimeout          = 1 * time.Minute
+)
+
+// NOTE: This test will mutate the default ingresscontroller
+func TestHealthCheckIntervalIngressController(t *testing.T) {
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, defaultName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	routerDeployment, err := getDeployment(t, kclient, controller.RouterDeploymentName(ic), 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get router deployment: %v", err)
+	}
+
+	// set spec.tuningOptions.healthCheckInterval to a nonstandard value, double the default
+	if err := setHealthCheckInterval(t, kclient, 10*time.Second, &metav1.Duration{2 * defaultHealthCheckInterval}, ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	if err := waitForIngressControllerCondition(t, kclient, 5*deploymentTimeout, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	if err := waitForDeploymentEnvVar(t, kclient, routerDeployment, deploymentTimeout, ingresscontroller.RouterBackendCheckInterval, (2 * defaultHealthCheckInterval).String()); err != nil {
+		t.Fatalf("expected router deployment to set %s to %v: %v", ingresscontroller.RouterBackendCheckInterval, 2*defaultHealthCheckInterval, err)
+	}
+
+	// set spec.healthCheckInterval to an unacceptable value, 1s
+	if err := setHealthCheckInterval(t, kclient, 1*time.Minute, &metav1.Duration{1 * time.Second}, ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	if err := waitForIngressControllerCondition(t, kclient, 5*deploymentTimeout, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// when set to an unacceptable value, it should remain as it was prior to the change
+	if err := waitForDeploymentEnvVar(t, kclient, routerDeployment, deploymentTimeout, ingresscontroller.RouterBackendCheckInterval, (2 * defaultHealthCheckInterval).String()); err != nil {
+		t.Fatalf("expected router deployment to keep %s set at %v: %v", ingresscontroller.RouterBackendCheckInterval, 2*defaultHealthCheckInterval, err)
```

Comment:
Why would setting an unacceptable value cause the operator to retain the previous setting?  I would expect the operator to revert to the default value.

Reply:
I remember discussing this and thought you told me it was ok to keep the old value.  Keeping the old value would make more sense to the user. Setting it to a seemingly random value if the new value was invalid seems unexpected to me, as a user.

Reply:
Fixed an error in the test logic.  Thanks!